***Chainguard: Notes on patch***

Zookeeper uses a version of netty which is affected by VE-2023-4586 and
CVE-2023-44487. We remediated these by bumping the netty version used in the
follwing PR: https://github.com/wolfi-dev/os/pull/30100/files.

Since then, zookeeper introduced a change in the 3.9.3 release:
 - https://github.com/apache/zookeeper/compare/release-3.9.2...release-3.9.3-0

Where they updated the sslHandler invocation to include a boolean argument:

```bash
SslHandler.isEncrypted(Unpooled.wrappedBuffer(litmus), false))
```

This causes a compliation issue with the latest netty version:

```
Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.8.1:compile (default-compile) on project zookeeper: Compilation failure
2024/10/07 14:34:09 INFO [ERROR] /home/build/zookeeper-server/src/main/java/org/apache/zookeeper/server/quorum/UnifiedServerSocket.java:[258,57] method isEncrypted in class io.netty.handler.ssl.SslHandler cannot be applied to given types;
2024/10/07 14:34:09 INFO [ERROR]   required: io.netty.buffer.ByteBuf
2024/10/07 14:34:09 INFO [ERROR]   found:    io.netty.buffer.ByteBuf,boolean
```

This patch remediates this, and allows us to continue to use the later netty
versions with the CVE fixes. Note the default is 'false', so the behavior is the
same.
-------
diff --git a/zookeeper-server/src/main/java/org/apache/zookeeper/server/quorum/UnifiedServerSocket.java b/zookeeper-server/src/main/java/org/apache/zookeeper/server/quorum/UnifiedServerSocket.java
index c101e9e2d..387a6c7d7 100644
--- a/zookeeper-server/src/main/java/org/apache/zookeeper/server/quorum/UnifiedServerSocket.java
+++ b/zookeeper-server/src/main/java/org/apache/zookeeper/server/quorum/UnifiedServerSocket.java
@@ -255,7 +255,7 @@ private void detectMode() throws IOException {
                 bytesRead = 0;
             }
 
-            if (bytesRead == litmus.length && SslHandler.isEncrypted(Unpooled.wrappedBuffer(litmus), false)) {
+            if (bytesRead == litmus.length && SslHandler.isEncrypted(Unpooled.wrappedBuffer(litmus))) {
                 try {
                     sslSocket = x509Util.createSSLSocket(prependableSocket, litmus);
                 } catch (X509Exception e) {
